helper.h

```diff
@@ -16,7 +16,7 @@ using namespace std;
 // the inf should be larger than any number in convex computation.
 // there are test case that y=x^2, and data size is 4e6
 // so the inf should be at least bigger than 1.6e13
-const double inf = 1e16;
+const double inf = 1e14;
```

Comment:
> there was some numerical problems because the number is to big but I don't know what the appropriate number should be

This note is outdated. Now our largest data set is 2^26, then the inf in parabolic should be bigger than 9*(2^26)^2, which is more than 1e16.
one solution is use different inf in different code. But it needs ton of test.